config/toml.go

```diff
@@ -550,4 +550,26 @@ max_open_connections = {{ .Instrumentation.MaxOpenConnections }}
 
 # Instrumentation namespace
 namespace = "{{ .Instrumentation.Namespace }}"
+
+#######################################################
+###       Data Companion Configuration Options          ###
+#######################################################
+[datacompanion]
+
+# Data Companion is enabled and we take the initial height as the official retain height:
```

Comment:
I don't understand this comment. I suspect `enabled` likely enables the DC, but as a user I might find the comment confusing.

Can we describe what `enabled` and `initial_retain_height` respectively do?

Reply:
I expanded on this in the ADR: https://github.com/cometbft/cometbft/blob/thane/adr-084-data-companion-pull-api/docs/architecture/adr-101-data-companion-pull-api.md#configuration

We'll need to update this configuration approach to match the ADR.